save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputTextFormRequired.kt

```diff
@@ -0,0 +1,149 @@
+     */
+    var form: InputTypes
+
+    /**
+     * Current input value
+     */
+    var textValue: String?
+
+    /**
+     * HTML bootstrap classes (className)
+     */
+    var classes: String
+
+    /**
+     * Label that will be displayed as an input title
+     */
+    var name: String
+
+    /**
+     * Flag that indicates if current [textValue] is valid or not
+     */
+    var validInput: Boolean
+
+    /**
+     * Conflict error message received from backend
+     */
+    var conflictMessage: String?
+
+    /**
+     * Callback invoked when a form is clicked
+     */
+    var onClickFun: () -> Unit
+
+    /**
+     * Callback invoked when [textValue] changed
+     */
+    var onChangeFun: (ChangeEvent<HTMLInputElement>) -> Unit
+}
+
+/**
+ * @param form
+ * @param validInput
+ * @param classes
+ * @param name
+ * @param onChangeFun
+ * @param textValue
+ * @param onClickFun
+ * @return div with an input form
+ */
+@Suppress(
+    "TOO_LONG_FUNCTION",
+    "TOO_MANY_PARAMETERS",
+    "LongParameterList",
+)
+private fun ChildrenBuilder.inputTextFormRequired(
+    form: InputTypes,
+    textValue: String?,
+    validInput: Boolean,
+    classes: String,
+    name: String,
+    conflictMessage: String?,
+    onClickFun: () -> Unit = { },
+    onChangeFun: (ChangeEvent<HTMLInputElement>) -> Unit = { }
+) {
+    div {
+        className = ClassName(classes)
+        label {
+            className = ClassName("form-label")
+            htmlFor = form.name
+            +name
+            span {
+                className = ClassName("text-danger text-left")
+                +"*"
+            }
+        }
+
+        div {
+            val inputType = if (form == InputTypes.PASSWORD) InputType.password else InputType.text
+            input {
+                type = inputType
+                onChange = onChangeFun
+                onClick = { onClickFun() }
+                id = form.name
+                required = true
+                value = textValue
+                placeholder = form.placeholder
+                className = if (textValue.isNullOrEmpty()) {
+                    ClassName("form-control")
+                } else if (validInput) {
+                    ClassName("form-control is-valid")
+                } else {
+                    ClassName("form-control is-invalid")
+                }
+                asDynamic()["data-toggle"] = "tooltip"
+                asDynamic()["data-placement"] = "right"
+                title = conflictMessage ?: form.tooltip
+            }
+            if (conflictMessage == null && !validInput && !textValue.isNullOrEmpty()) {
+                div {
+                    className = ClassName("invalid-feedback d-block")
+                    +(form.errorMessage ?: "Input a valid ${form.str}, please.")
+                }
+            }
+        }
+    }
+}
+
+private fun inputTextFormRequiredWrapper() = FC<InputTextFormRequiredProps> { props ->
+    useTooltip()
+    inputTextFormRequired(
```

Comment:
Why do you need a separate function? Maybe create elements right here, because otherwise you have to pass too many parameters

Reply:
@akuleshov7 once mentioned that this is a temporary solution